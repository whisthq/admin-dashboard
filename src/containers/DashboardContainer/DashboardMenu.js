import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { withRouter } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { Dropdown } from 'react-bootstrap'
import {
    faHome,
    faChartArea,
    faList,
    faExternalLinkAlt,
} from '@fortawesome/free-solid-svg-icons'
import { FaLink } from 'react-icons/fa'

import 'static/App.css'

import Logo from 'assets/logo.svg'

withRouter((props) => <DashboardMenu {...props} />)

class DashboardMenu extends Component {
    constructor(props) {
        super(props)
        this.state = { page: 'dashboard' }
    }

    render() {
        return (
            <div
                style={{
                    backgroundColor: 'white',
                    height: '100%',
                    width: 75,
                    minHeight: '100vh',
                    paddingRight: 25,
                    paddingLeft: 0,
                    paddingTop: 50,
                }}
            >
                <div style={{ paddingLeft: 25 }}>
                    <img
                        src={Logo}
                        alt=""
                        style={{ width: 30, height: 30, marginBottom: 70 }}
                    />
                </div>
                <div style={{ marginTop: 40, fontSize: 18 }}>
                    <div style={{ marginBottom: 30, marginLeft: 5 }}>
                        <Link
                            style={{
                                textDecoration: 'none',
                                color:
                                    this.props.page === 'dashboard'
                                        ? '#111111'
                                        : '#999999',
                                fontWeight:
                                    this.props.page === 'dashboard'
                                        ? 'bold'
                                        : 'normal',
                                borderLeft:
                                    this.props.page === 'dashboard'
                                        ? 'solid 4px #6241e8'
                                        : 'solid 4px white',
                                paddingLeft: 18,
                            }}
                            to="/admin/dashboard"
                        >
                            <FontAwesomeIcon
                                icon={faHome}
                                style={{
                                    color:
                                        this.props.page === 'dashboard'
                                            ? '#111111'
                                            : '#d0ced9',
                                    fontSize: 20,
                                }}
                            />
                        </Link>
                    </div>
                    <div style={{ marginBottom: 30, marginLeft: 5 }}>
                        <Link
                            style={{
                                textDecoration: 'none',
                                fontWeight:
                                    this.props.page === 'analytics'
                                        ? 'bold'
                                        : 'normal',
                                borderLeft:
                                    this.props.page === 'analytics'
                                        ? 'solid 4px #6241e8'
                                        : 'solid 4px white',
                                paddingLeft: 20,
                            }}
                            to="/admin/analytics"
                        >
                            <FontAwesomeIcon
                                icon={faChartArea}
                                style={{
                                    color:
                                        this.props.page === 'analytics'
                                            ? '#111111'
                                            : '#d0ced9',
                                    fontSize: 20,
                                }}
                            />
                        </Link>
                    </div>
                    <div style={{ marginBottom: 30, marginLeft: 5 }}>
                        <Link
                            style={{
                                textDecoration: 'none',
                                fontWeight:
                                    this.props.page === 'logs'
                                        ? 'bold'
                                        : 'normal',
                                borderLeft:
                                    this.props.page === 'logs'
                                        ? 'solid 4px #6241e8'
                                        : 'solid 4px white',
                                paddingLeft: 21,
                            }}
                            to="/admin/logs"
                        >
                            <FontAwesomeIcon
                                icon={faList}
                                style={{
                                    color:
                                        this.props.page === 'logs'
                                            ? '#111111'
                                            : '#d0ced9',
                                    fontSize: 19,
                                }}
                            />
                        </Link>
                    </div>
                    <Dropdown
                        style={{
                            textDecoration: 'none',
                            paddingLeft: 29,
                        }}
                    >
                        <Dropdown.Toggle variant="success" id="dropdown-basic">
                            <FontAwesomeIcon
                                icon={faExternalLinkAlt}
                                style={{
                                    color: '#d0ced9',
                                    fontSize: 18,
                                    outline: 'none',
                                }}
                            />
                        </Dropdown.Toggle>

                        <Dropdown.Menu style={{ border: 'none', padding: 20 }}>
                            <Dropdown.Item style={{ marginBottom: 5 }}>
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://s3.console.aws.amazon.com/s3/home?region=us-east-1#"
                                >
                                    <FaLink
                                        style={{
                                            marginRight: 10,
                                            color: '#d0ced9',
                                            fontSize: 12,
                                        }}
                                    />
                                    AWS S3
                                </a>
                            </Dropdown.Item>{' '}
                            <Dropdown.Item style={{ marginBottom: 5 }}>
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://lightsail.aws.amazon.com/ls/webapp/home/instances?#"
                                >
                                    <FaLink
                                        style={{
                                            marginRight: 10,
                                            color: '#d0ced9',
                                            fontSize: 12,
                                        }}
                                    />
                                    AWS Lightsail
                                </a>
                            </Dropdown.Item>
                            <Dropdown.Item style={{ marginBottom: 5 }}>
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://dashboard.heroku.com/teams/fractalcomputers/apps"
                                >
                                    <FaLink
                                        style={{
                                            marginRight: 10,
                                            color: '#d0ced9',
                                            fontSize: 12,
                                        }}
                                    />
                                    Heroku
                                </a>
                            </Dropdown.Item>
                            <Dropdown.Item style={{ marginBottom: 5 }}>
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://fractalcomputers.postman.co/workspaces/6ded464f-396a-4792-a5c9-5ac7c689d269/apis"
                                >
                                    <FaLink
                                        style={{
                                            marginRight: 10,
                                            color: '#d0ced9',
                                            fontSize: 12,
                                        }}
                                    />
                                    Postman
                                </a>
                            </Dropdown.Item>
                            <Dropdown.Item style={{ marginBottom: 5 }}>
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://drive.google.com/open?id=1kbpGbVeYd5LVJgzGN0dKro-ryizQr2kg"
                                >
                                    <FaLink
                                        style={{
                                            marginRight: 10,
                                            color: '#d0ced9',
                                            fontSize: 12,
                                        }}
                                    />
                                    Google Drive
                                </a>
                            </Dropdown.Item>
                            <Dropdown.Item style={{ marginBottom: 5 }}>
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://www.notion.so/fractalcomputers/2fa59d87701b4ec7967377450c75008a?v=f20e6d7d6d5f4d4c9a0f886871259833"
                                >
                                    <FaLink
                                        style={{
                                            marginRight: 10,
                                            color: '#d0ced9',
                                            fontSize: 12,
                                        }}
                                    />
                                    Notion
                                </a>
                            </Dropdown.Item>
                        </Dropdown.Menu>
                    </Dropdown>
                </div>
            </div>
        )
    }
}

function mapStateToProps(state) {
    return {
        page: state.AccountReducer.page,
    }
}

export default connect(mapStateToProps)(DashboardMenu)
