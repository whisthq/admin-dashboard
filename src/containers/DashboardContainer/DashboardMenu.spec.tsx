import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { DashboardMenu } from './DashboardMenu'
import { Link } from 'react-router-dom'
import { Dropdown } from 'react-bootstrap'
import toJson from 'enzyme-to-json'

// necessary for enzyme to work
Enzyme.configure({ adapter: new Adapter() })

/*
 */
describe('<DashboardMenu />', () => {
    // snapshot test
    it('renders correctly', () => {
        const wrapper = shallow(
            <DashboardMenu
                page="dashboard"
                dispatch={(items: any) => {
                    // mock
                }}
            />
        )

        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders three <Link /> components', () => {
        const wrapper = shallow(
            <DashboardMenu
                page="dashboard"
                dispatch={(items: any) => {
                    // mock
                }}
            />
        )
        expect(wrapper.find(Link)).toHaveLength(3)
    })

    it('renders one <Dropdown /> component with a toggle and a menu with eight items', () => {
        const wrapper = shallow(
            <DashboardMenu
                page="dashboard"
                dispatch={(items: any) => {
                    // mock
                }}
            />
        )
        expect(wrapper.find(Dropdown)).toHaveLength(1)

        expect(wrapper.find(Dropdown.Toggle)).toHaveLength(1)
        expect(wrapper.find(Dropdown.Menu)).toHaveLength(1)

        // should be s3, lightsail, heroku, datadog, sentry, postman, drive, notion
        expect(wrapper.find(Dropdown.Item)).toHaveLength(8)
    })

    it('renders four <FontAwesomeIcon /> with the correct picture for the links/dropdown', () => {
        const wrapper = shallow(
            <DashboardMenu
                page="dashboard"
                dispatch={(items: any) => {
                    // mock
                }}
            />
        )
        expect(wrapper.find(FontAwesomeIcon)).toHaveLength(4)
    })
})
