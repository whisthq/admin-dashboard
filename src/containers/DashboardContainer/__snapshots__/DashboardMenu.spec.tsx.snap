// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardMenu /> renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "height": "100%",
      "minHeight": "100vh",
      "paddingLeft": 0,
      "paddingRight": 25,
      "paddingTop": 50,
      "width": 75,
    }
  }
>
  <div
    style={
      Object {
        "paddingLeft": 25,
      }
    }
  >
    <img
      alt=""
      src="logo.svg"
      style={
        Object {
          "height": 30,
          "marginBottom": 70,
          "width": 30,
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "fontSize": 18,
        "marginTop": 40,
      }
    }
  >
    <div
      style={
        Object {
          "marginBottom": 30,
          "marginLeft": 5,
        }
      }
    >
      <Link
        style={
          Object {
            "borderLeft": "solid 4px #6241e8",
            "color": "#111111",
            "fontWeight": "bold",
            "paddingLeft": 18,
            "textDecoration": "none",
          }
        }
        to="/admin/dashboard"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                576,
                512,
                Array [],
                "f015",
                "M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z",
              ],
              "iconName": "home",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          style={
            Object {
              "color": "#111111",
              "fontSize": 20,
            }
          }
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </Link>
    </div>
    <div
      style={
        Object {
          "marginBottom": 30,
          "marginLeft": 5,
        }
      }
    >
      <Link
        style={
          Object {
            "borderLeft": "solid 4px white",
            "fontWeight": "normal",
            "paddingLeft": 20,
            "textDecoration": "none",
          }
        }
        to="/admin/analytics"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f1fe",
                "M500 384c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H12c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v308h436zM372.7 159.5L288 216l-85.3-113.7c-5.1-6.8-15.5-6.3-19.9 1L96 248v104h384l-89.9-187.8c-3.2-6.5-11.4-8.7-17.4-4.7z",
              ],
              "iconName": "chart-area",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          style={
            Object {
              "color": "#d0ced9",
              "fontSize": 20,
            }
          }
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </Link>
    </div>
    <div
      style={
        Object {
          "marginBottom": 30,
          "marginLeft": 5,
        }
      }
    >
      <Link
        style={
          Object {
            "borderLeft": "solid 4px white",
            "fontWeight": "normal",
            "paddingLeft": 21,
            "textDecoration": "none",
          }
        }
        to="/admin/logs"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f03a",
                "M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z",
              ],
              "iconName": "list",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          style={
            Object {
              "color": "#d0ced9",
              "fontSize": 19,
            }
          }
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </Link>
    </div>
    <Dropdown
      navbar={false}
      style={
        Object {
          "paddingLeft": 29,
          "textDecoration": "none",
        }
      }
    >
      <DropdownToggle
        id="dropdown-basic"
        variant="success"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f35d",
                "M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z",
              ],
              "iconName": "external-link-alt",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          style={
            Object {
              "color": "#d0ced9",
              "fontSize": 18,
              "outline": "none",
            }
          }
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </DropdownToggle>
      <DropdownMenu
        alignRight={false}
        flip={true}
        style={
          Object {
            "border": "none",
            "padding": 20,
          }
        }
      >
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
          style={
            Object {
              "marginBottom": 5,
            }
          }
        >
          <a
            href="https://s3.console.aws.amazon.com/s3/home?region=us-east-1#"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FaLink
              style={
                Object {
                  "color": "#d0ced9",
                  "fontSize": 12,
                  "marginRight": 10,
                }
              }
            />
            AWS S3
          </a>
        </DropdownItem>
         
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
          style={
            Object {
              "marginBottom": 5,
            }
          }
        >
          <a
            href="https://lightsail.aws.amazon.com/ls/webapp/home/instances?#"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FaLink
              style={
                Object {
                  "color": "#d0ced9",
                  "fontSize": 12,
                  "marginRight": 10,
                }
              }
            />
            AWS Lightsail
          </a>
        </DropdownItem>
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
          style={
            Object {
              "marginBottom": 5,
            }
          }
        >
          <a
            href="https://dashboard.heroku.com/teams/fractalcomputers/apps"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FaLink
              style={
                Object {
                  "color": "#d0ced9",
                  "fontSize": 12,
                  "marginRight": 10,
                }
              }
            />
            Heroku
          </a>
        </DropdownItem>
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
          style={
            Object {
              "marginBottom": 5,
            }
          }
        >
          <a
            href="https://app.datadoghq.com/logs?cols=core_host%2Ccore_service&from_ts=1593977274176&index=&live=true&messageDisplay=inline&stream_sort=desc&to_ts=1593978174176"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FaLink
              style={
                Object {
                  "color": "#d0ced9",
                  "fontSize": 12,
                  "marginRight": 10,
                }
              }
            />
            Datadog
          </a>
        </DropdownItem>
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
          style={
            Object {
              "marginBottom": 5,
            }
          }
        >
          <a
            href="https://sentry.io/organizations/fractalcomputers/projects/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FaLink
              style={
                Object {
                  "color": "#d0ced9",
                  "fontSize": 12,
                  "marginRight": 10,
                }
              }
            />
            Sentry
          </a>
        </DropdownItem>
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
          style={
            Object {
              "marginBottom": 5,
            }
          }
        >
          <a
            href="https://fractalcomputers.postman.co/workspaces/6ded464f-396a-4792-a5c9-5ac7c689d269/apis"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FaLink
              style={
                Object {
                  "color": "#d0ced9",
                  "fontSize": 12,
                  "marginRight": 10,
                }
              }
            />
            Postman
          </a>
        </DropdownItem>
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
          style={
            Object {
              "marginBottom": 5,
            }
          }
        >
          <a
            href="https://drive.google.com/open?id=1kbpGbVeYd5LVJgzGN0dKro-ryizQr2kg"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FaLink
              style={
                Object {
                  "color": "#d0ced9",
                  "fontSize": 12,
                  "marginRight": 10,
                }
              }
            />
            Google Drive
          </a>
        </DropdownItem>
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
          style={
            Object {
              "marginBottom": 5,
            }
          }
        >
          <a
            href="https://www.notion.so/fractalcomputers/2fa59d87701b4ec7967377450c75008a?v=f20e6d7d6d5f4d4c9a0f886871259833"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FaLink
              style={
                Object {
                  "color": "#d0ced9",
                  "fontSize": 12,
                  "marginRight": 10,
                }
              }
            />
            Notion
          </a>
        </DropdownItem>
      </DropdownMenu>
    </Dropdown>
  </div>
</div>
`;
